crates/wasmparser/src/readers/core/types.rs

```diff
@@ -510,6 +531,17 @@ impl RefType {
     }
 }
 
+impl Matches for RefType {
+    fn matches<'a, F>(&self, other: &Self, type_at: &F) -> Result<bool>
+    where
+        F: Fn(u32) -> Result<&'a SubType>,
+    {
+        Ok((*self == *other)
+            || (other.is_nullable() || !self.is_nullable())
+                && self.heap_type().matches(&other.heap_type(), type_at)?)
```

Comment:
Could parentheses be used to clarify the order of operations here? I don't know the binding precedence of `||` vs `&&` myself

Reply:
Sure thing!

And I thought people didn't like parentheses. Lisp is the best.